examples/Tests/ParametricEq.hpp

```diff
@@ -0,0 +1,585 @@
+           ...);
+        },
+        std::make_tuple(
+            std::tie(
+                inputs.band1,
+                inputs.toggle1,
+                inputs.order1,
+                inputs.cutoffFreq1,
+                inputs.centerFreq1,
+                inputs.gain1,
+                inputs.freqBandWidth1,
+                inputs.passbandRipple1),
+            std::tie(
+                inputs.band2,
+                inputs.toggle2,
+                inputs.order2,
+                inputs.cutoffFreq2,
+                inputs.centerFreq2,
+                inputs.gain2,
+                inputs.freqBandWidth2,
+                inputs.passbandRipple2),
+            std::tie(
+                inputs.band3,
+                inputs.toggle3,
+                inputs.order3,
+                inputs.cutoffFreq3,
+                inputs.centerFreq3,
+                inputs.gain3,
+                inputs.freqBandWidth3,
+                inputs.passbandRipple3),
+            std::tie(
+                inputs.band4,
+                inputs.toggle4,
+                inputs.order4,
+                inputs.cutoffFreq4,
+                inputs.centerFreq4,
+                inputs.gain4,
+                inputs.freqBandWidth4,
+                inputs.passbandRipple4),
+            std::tie(
+                inputs.band5,
+                inputs.toggle5,
+                inputs.order5,
+                inputs.cutoffFreq5,
+                inputs.centerFreq5,
+                inputs.gain5,
+                inputs.freqBandWidth5,
+                inputs.passbandRipple5),
+            std::tie(
+                inputs.band6,
+                inputs.toggle6,
+                inputs.order6,
+                inputs.cutoffFreq6,
+                inputs.centerFreq6,
+                inputs.gain6,
+                inputs.freqBandWidth6,
+                inputs.passbandRipple6),
+            std::tie(
+                inputs.band7,
+                inputs.toggle7,
+                inputs.order7,
+                inputs.cutoffFreq7,
+                inputs.centerFreq7,
+                inputs.gain7,
+                inputs.freqBandWidth7,
+                inputs.passbandRipple7),
+            std::tie(
+                inputs.band8,
+                inputs.toggle8,
+                inputs.order8,
+                inputs.cutoffFreq8,
+                inputs.centerFreq8,
+                inputs.gain8,
+                inputs.freqBandWidth8,
+                inputs.passbandRipple8)));
+    if(toRecompute)
+    {
+      // computes the new bode diagram for the equalizer
+      std::cout << "about to kill your machine lol " << std::endl;
+      diagramCalculator.computeDiagram();
+      // sends a message to the ui to update the graph
+      processor_to_ui msg;
+      msg.xInput = diagramCalculator.frequencies.data();
+      msg.yInput = diagramCalculator.responses.data();
+      send_message(std::move(msg));
+      toRecompute = false;
```

Comment:
one thing that could be done is to recompute it only once in a while, e.g. every 3 buffers or 15 milliseconds. Or compute it on the UI side maybe.

Reply:
right now, it is only recomputed when a filter parameter is changed which I believe is the most efficient solution (there's no need to recompute the graph if none of the filters changes, which, if we were to recompute every once in a while would happen)